### DISCUSSION TITLE
Why code reviewer for branch protection must have write permission

### DISCUSSION THREAD

**Discussion by leslie-qiwa** [community/community#18067] [Pull Requests]
My issue is well explained in https://stackoverflow.com/questions/62318630/require-pr-review-in-branch-protection-rules-without-write-access.

From https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-pull-request-reviews-before-merging, I know code reviewer for branch protection must have write permission, but just don't understand why it is designed this way. Can anyone please explain the rational behind?

**Comment by wuwentao** [comment]
we also have this issue and submit a github support ticket **#1647840** ,  seems they can't help with it.

This is our Github feature design issue, I can't found any option or documents for this write permission required setting.
we(Github) should have a switch option to [**enable or disable**] code review required write access for this.

in additional,  code owner is a fixed solution and we don't want to use it.
if the code reviewer have write permission, they also can merge PR to branch, this is the root cause, code review owner don't need
 write(merge PR) permission, only need code review approve permssion.
